### ISSUE TITLE
chainit 3: why a state machine and can it be externalized

### ISSUE THREAD

**Issue by vvo** [vvo/chainit#16]
Are you planning on doing it @kapouer?

I guess we need to decide on error handling otherwise we would introduce another breaking change

**Comment by vvo** [comment]
I really do not understand the state machine thingy. Why we need it in core.

As you saw with the webdriver example, it seems we should try to keep chainit as small as possible and do not introduce features that are only needed by some users.

Maybe a plugin system would be enough.

``` js
chainit.use(require('chainit-state'));
```

?

**Comment by kapouer** [comment]
Yes i'm planning to do it, but i'd like to address the two issues you raised before.
I pushed chainit-3 branch and will merge it to master when those issues are dealt with.
Are you ok with that ?

**Comment by vvo** [comment]
Yes no problem!
